### DISCUSSION TITLE
🐛 All themes look the same

### DISCUSSION THREAD

**Discussion by esheehan-gsl** [dandavison/delta#1396] [General]
When I run `delta --show-syntax-themes --dark`, all of the themes look the same.

![Screenshot of Dracula and Monokai Extended theme samples, both of which are just identical green and red](https://user-images.githubusercontent.com/85259329/198059493-cbb60b1e-ce40-4921-b024-ebf6b99308de.png)

- **Terminal**: Alacritty 0.10.1-rc1
- **Operating System**: Pop!_OS (Ubuntu) 22.04
- **Install method**: git-delta_0.14.0_amd64.deb

For reference, I have installed `bat` from the official Pop!_OS apt sources, and the themes all work there. I have also tested `delta` in GNOME Terminal 3.44.0 and the output is the same red and green that I get in Alacritty.

- [ ] Please include the raw text output from git, so that we can reproduce the problem.
      (You can use `git --no-pager` to produce the raw text output.)
- [x] A screenshot of Delta's output is often helpful also.

Thanks for filing a Delta bug report!

**Comment by magnus-ISU** [comment]
They clearly don't?

On the top, the green comment text is `#5862A2` and on the bottom `#4E684E` for example.

**Comment by bizrad** [comment]
I'm seeing the same issue with `--show-syntax-themes` delta 0.15.1 and doing a fresh build from HEAD @ 8fd6486, on MacOS 13.3.1 M2 using iTerm2 3.4.19 with `COLORTERM=truecolor` and bat 0.23.0. I also see the same behavior on the Apple Terminal. Using the syntax theme in bat looks as expected, but I can't get delta to change themes using the what I've found in the documentation.

**Comment by dandavison** [comment]
I've converted this to a discussion because I don't think there's a delta bug -- I suspect this is problems with individual environments.

`delta --show-syntax-themes` works for me with delta 0.15.1 from Homebrew.

@bizrad if you do `git show | delta --syntax-theme=$SOME_SYNTAX_THEME` for a few different themes does that seem to work correctly? Can you post screenshots if not?

**Reply by bizrad** [reply]
Ok, so here's starting a new shell to clear out anything from `.zshrc` and while they do look very similar you can see there are also slight differences. However, they clearly also aren't the version seen from `bat` so this makes me think it's doing some fallback mode from truecolor or something.

Let me know if you have any ideas on how I can troubleshoot or get debugging information to share.

```
custom-shell $ sw_vers
ProductName:		macOS
ProductVersion:		13.3.1
BuildVersion:		22E261
custom-shell $ zsh -f
My-MacBook-Pro% PROMPT='%F{033}[%*]:%~ $%F{reset} '
[14:56:33]:~ $ cd tmp-repo
[14:56:49]:~/tmp-repo $ env|grep TERM|grep -v SESSION
LC_TERMINAL_VERSION=3.4.19
ITERM_PROFILE=Default
TERM_PROGRAM_VERSION=3.4.19
TERM_PROGRAM=iTerm.app
LC_TERMINAL=iTerm2
COLORTERM=truecolor
TERM=xterm-256color
[14:57:10]:~/tmp-repo $ delta --version
delta 0.15.1
[14:57:17]:~/tmp-repo $ bat --version
bat 0.23.0
```
![image](https://user-images.githubusercontent.com/7604131/232495269-75566c0e-ed95-4392-b7f4-01f535b4e5a3.png)
![image](https://user-images.githubusercontent.com/7604131/232496351-dfc3233a-3566-4c2f-9f56-d34f7c142172.png)
![image](https://user-images.githubusercontent.com/7604131/232496923-fcbc35da-4c7c-416b-b91c-4ec42cea9497.png)

**Comment by dandavison** [comment]
OK would you mind spelling it out for me -- what's an example of a bug here? For example Nord looks identical between Delta and Bat doesn't it?

**Reply by esheehan-gsl** [reply]
The issue is that the red and green used to highlight additions and subtractions in the diff is the same no matter what syntax theme you choose. I would expect that for any theme, a shade of red and green are chosen from that theme’s colors.

**Reply by dandavison** [reply]
OK, that expectation's not actually correct. The "syntax theme" refers specifically to the foreground colors used for syntax highlighting, as opposed to background colors used for diff highlighting. That's the reason the option has that slightly pedantic name "syntax theme" as opposed to just "theme". But delta does also have an informal concept of theme including both foreground and background colors.

The documentation on all this is here: https://dandavison.github.io/delta/custom-themes.html

**Reply by bizrad** [reply]
Ahh ok, I misunderstood the docs. I didn't want to make a "custom theme" I wanted to use a prebuilt one. Perhaps they could be a bit clearer and include instructions on using the themes in the main configuration section. I read I can use the same themes as `bat` thought I could simply reference "Nord" somewhere.

**Reply by dandavison** [reply]
Right, you can reference "Nord" as `--syntax-theme=Nord`. But you'll still have to choose background colors for the diff highlighting since Nord doesn't specify those (and IIRC bat promises never to alter background colors).

**Comment by bizrad** [comment]
Here are the full step-by-step commands to set this up on a mac in case anyone else is confused by this. If you just want to add the default themes use the last 3 commands.
```
brew install git-delta
# default settings
git config --global core.pager "delta"
git config --global interactive.diffFilter "delta --color-only"
git config --global delta.navigate "true"
git config --global delta.light "false"
git config --global merge.conflictstyle "diff3"
git config --global diff.colorMoved "default"

#Optional
git config --global side-by-side "true"

#Theme setup 
wget --output-document ~/delta-themes.gitconfig https://raw.githubusercontent.com/dandavison/delta/master/themes.gitconfig
git config --global include.path  "~/delta-themes.gitconfig"
git config --global delta.features "collared-trogon"
```
